src/construction/identity_based_encryption/dual_regev_ibe.rs

```diff
@@ -345,7 +349,7 @@ impl IBE for DualRegevIBE {
         let secret_key = self.psf.samp_p(master_pk, master_sk, &u);
 
         // insert secret key in HashMap
-        self.storage.insert(identity.clone(), secret_key.clone()); //todo insert for different pk and sk
+        self.storage.insert(identity.clone(), secret_key.clone());
```

Comment:
Just insert a tuple `(master_pk, master_sk, identity)` as key into the hash map (or if this is getting too big, use a hashed version of the tuple) to get rid of the WARNING.

Reply:
applyed